### PR TITLE
Expand Boolean API convention example

### PR DESCRIPTION
**Pull request by JoelSpeed**
Some additional feedback came in after #1215 merged, this PR addresses that feedback

CC @Miciah

### CODE REVIEW

File: dev-guide/api-conventions.md [hunk 1]
```diff
@@ -465,14 +465,14 @@ the resource group and name that identify the resource.
...
-As a Boolean value can only ever have 3 values (`true`, `false`, `omitted` when a pointer), we have seen examples in
-where API authors have later added additional fields, paired with the Boolean field that are only meaningful when the
-original field has a certain state. This makes it confusing for an end user as they have to be aware that the field
-they are trying to use, only has an effect in certain circumstances.
-
-Rather than creating a boolean field:
+Many ideas start as a Boolean value, e.g. `FooEnabled: true|false`, but often evolve into needing 3, 4, or even more
+states at some point during the API's lifetime.
+As a Boolean value can only ever have 2 or in some cases 3 values (`true`, `false`, `omitted` when a pointer), we have
+seen examples in which API authors have later added additional fields, paired with a Boolean field that are only
```
Comment (Miciah) [inline dev-guide/api-conventions.md:471]:
```suggestion
seen examples in which API authors have later added additional fields, paired with a Boolean field, that are only
```

File: dev-guide/api-conventions.md [hunk 2]
```diff
@@ -465,14 +465,14 @@ the resource group and name that identify the resource.
...
+meaningful when the original field has a certain state. This makes it confusing for an end user as they have to be
+aware that the field they are trying to use, only has an effect in certain circumstances.
```
Comment (Miciah) [inline dev-guide/api-conventions.md:473]:
```suggestion
aware that the field they are trying to use only has an effect in certain circumstances.
```

File: dev-guide/api-conventions.md [hunk 3]
```diff
@@ -482,16 +482,20 @@ AuthenticationEnabled *bool `json:"authenticationEnabled,omitempty"`
 
 Use an enumeration of values that describe the action instead:
 ```go
-// authentication determines the requirements for authentication within the cluster.
-// When omitted, the authentication will be Optional.
+// authentication determines the requirements for authentication within the cluster. Allowed values are "Optional",
+// "Required", "Disabled" and omitted.
+// Optional authentication allows users to optionally authenticate but will not reject an unathenticated request.
+// Required authentication requires all requests to be authenticated.
+// Disabled authentication ignores any attempt to authenticate and proccesses all requests as unathenticated.
```
Comment (Miciah) [inline dev-guide/api-conventions.md:489]:
```suggestion
// Optional authentication allows users to optionally authenticate but will not reject an unauthenticated request.
// Required authentication requires all requests to be authenticated.
// Disabled authentication ignores any attempt to authenticate and processes all requests as unauthenticated.
```

**Comment by JoelSpeed** [comment]
Thanks for the review, have addressed those three suggestions

**Pull request merged**